src/main/webapp/src/app/admin-page/admin-page.component.ts

```diff
@@ -0,0 +1,341 @@
+   */
+  getUserInfo(user: string): Observable<HttpResponse<User>> {
+    return this.httpClient.get<User>('/user/' + user, {
+      observe: 'response',
+      responseType: 'json',
+    });
+  }
+
+  /**
+   * Gets plant information for the specified plant from
+   * the server. Returns an observable HTTP response.
+   *
+   * Performs GET: /plant/{plant}
+   *
+   * @param plant The plant reqested from the server.
+   * @return the http response.
+   */
+  getPlantInfo(plant: string): Observable<HttpResponse<Plant>> {
+    return this.httpClient.get<Plant>('/plant/' + plant, {
+      observe: 'response',
+      responseType: 'json',
+    });
+  }
+
+  /**
+   * Creates garden summary.
+   *
+   * @param garden The garden to create an admin page of.
+   */
+  createGardenSummary(garden: string): void {
+    this.getGardenInfo(garden).subscribe({
+      next: (response: HttpResponse<Garden>) => {
+        // Successful responses are handled here.
+        this.gardenProfile = response.body;
+        this.isLoaded = true;
+        // Get rest of information
+        this.createGardenUserList(garden);
+        this.createGardenPlantList(garden);
+        this.gardenManager = 'Loading';
+        this.getUserInfo(this.gardenProfile.adminId).subscribe({
+          next: (response: HttpResponse<User>) => {
+            this.gardenManager = response.body.preferredName;
+          },
+          error: (error: HttpErrorResponse) => {
+            // Handle connection error
+            if (error.error instanceof ErrorEvent) {
+              console.error('Network error: ' + error.error.message);
+              this.gardenManager = 'Cannot fetch name';
+              return;
+            }
+            console.error('Unexpected error: ' + error.statusText);
+            this.gardenManager = 'Cannot fetch name';
+          },
+        });
+      },
+      error: (error: HttpErrorResponse) => {
+        // Handle connection error
+        if (error.error instanceof ErrorEvent) {
+          console.error('Network error: ' + error.error.message);
+          this.gardenProfile = null;
+          this.errorMessage = 'Cannot connect to GrowPod Server';
+          this.isLoaded = true;
+          return;
+        }
+        // Non-404 error codes
+        if (error.status !== 404) {
+          console.error('Unexpected error: ' + error.statusText);
+          this.gardenProfile = null;
+          this.errorMessage =
+            'Unexpected error ' + error.status + ': ' + error.statusText;
+          this.isLoaded = true;
+          return;
+        }
+        console.error('Error ' + error.status + ': ' + error.statusText);
+        this.gardenProfile = null;
+        this.errorMessage =
+          'Cannot see garden profile for garden id: ' + garden;
+        this.isLoaded = true;
+      },
+    });
+  }
+
+  /**
+   * Creates garden user list.
+   *
+   * @param garden The garden to create a user list of.
+   */
+  createGardenUserList(garden: string): void {
```

Comment:
does this create anything on the backend or does the `create` reference creating UI elements?

Reply:
`Create` refers to UI elements, but I can make this clearer in the comments